mtwitter.js

```diff
@@ -0,0 +1,254 @@
+            console.log("Checking cache for @" + name);
+            var cached = this.tuser_cache[name];
+            if (ts - cached.cache_time < TWITTER_PROFILE_INTERVAL) {
+                resolve(cached.user);
+            }
+        }
+
+        console.log("Checking twitter for @" + name);
+        this.app_twitter.get('users/show', {
+            screen_name: name
+        }, (error, user, response) => {
+            if (error) {
+                this.tuser_cache[name] = {
+                    cache_time: ts,
+                    user: null
+                };
+                reject(error);
+            }
+            this.tuser_cache[name] = {
+                cache_time: ts,
+                user: user
+            };
+            resolve(user);
+        });
+    });
+
+}
+
+/*
+  Add a user's timeline to the iterator. Tweets will be automatically send to
+  the given room.
+*/
+mtwitter.prototype.enqueue_timeline = function(userid, localroom, remoteroom) {
+    var obj = {
+        "user_id": userid,
+        "local": localroom,
+        "remote": remoteroom
+    };
+    this.timeline_list.push(obj);
+    this.timeline_queue.push(obj);
+}
+
+/*
+  This function will fill the content structure for a new matrix message
+  for a given tweet.
+*/
+mtwitter.prototype.construct_message = function(tweet, type) {
+    return {
+        "body": tweet.text,
+        "created_at": tweet.created_at,
+        "likes": tweet.favorite_count,
+        "reblogs": tweet.retweet_count,
+        "tweet_id": tweet.id_str,
+        "tags": tweet.entities.hashtags,
+        "msgtype": type
+    }
+}
+
+/*
+  Process a given tweet (including resolving any parent parent tweets), and
+  submit it to the appropriate room.
+*/
+mtwitter.prototype.process_tweet = function(bridge, roomid, tweet, treeN) {
+    //console.log(tweet);
+    var muser = "@twitter_" + tweet.user.id_str + ":" + bridge.opts.domain;
+    var intent = bridge.getIntent(muser);
+    treeN--;
+    if (treeN < 0) {
+        console.log("Bailing because we have gone too far deep.")
+        return;
+    }
+
+    var type = "m.text";
+    if (tweet.in_reply_to_status_id_str != null) {
+        type = "m.notice"; // A nicer way to show previous tweets
+    }
+    var tweet_content = this.construct_message(tweet, type);
+
+    if (tweet.in_reply_to_status_id_str != null) {
+        this.app_twitter.get('statuses/show/' + tweet.in_reply_to_status_id_str, {}, (error, newtweet, response) => {
+            if (!error) {
+                this.process_tweet(bridge, roomid, newtweet, treeN);
+                return;
+            }
+            console.error(error);
+        });
+        setTimeout(() => {
+            intent.sendEvent(roomid, "m.room.message", tweet_content)
+        }, treeN * 250); //Make sure not to send them too quickly.
+    }
+}
+
+/*
+  Internal function to process the timeline queue.
+*/
+mtwitter.prototype._process_timeline = function(self) {
```

Comment:
Why are you passing `self` in here? This isn't python. You have an implicit `this`.